matchzoo/generators/point_generator.py

```diff
@@ -68,25 +67,24 @@ def _get_batch_of_transformed_samples(self, index_array: list):
         :return: A batch of transformed samples.
 
         """
-        batch_size = len(index_array)
-        batch_x_left = []
-        batch_x_right = []
-        batch_ids = []
-        if isinstance(self._task, tasks.Ranking):
-            batch_y = self.y
-        elif isinstance(self._task, tasks.Classification):
-            batch_y = np.zeros((batch_size, self._task._num_classes),
-                               dtype=np.int32)
-            for i, label in enumerate(self.y[index_array]):
-                batch_y[i, label] = 1
-        else:
-            raise ValueError('Error target mode in point generator.')
-
-        for key, val in enumerate(index_array):
-            batch_x_left.append(self.x_left[val])
-            batch_x_right.append(self.x_right[val])
-            batch_ids.append(self.ids[val])
-
-        return ({'x_left': np.array(batch_x_left), 'x_right':
-                 np.array(batch_x_right), 'ids':
-                 batch_ids}, np.array(batch_y))
+        bsize = len(index_array)
+        batch_x = {}
+        batch_y = None
+        if 'label' in self.data:
+            if isinstance(self._task, tasks.Ranking):
+                batch_y = self.data['label']
+            elif isinstance(self._task, tasks.Classification):
+                batch_y = np.zeros((bsize, self._task._num_classes),
```

Comment:
Use `num_classes` instead of `_num_classes`, `num_classes` is a property, see definition:

https://github.com/faneshion/MatchZoo/blob/8ca2b0e9920cfad67c287a5df43c1d64f4ce5d3a/matchzoo/tasks/classification.py#L18-L21